src/routes/api/client-key/client-key-regex.ts

```diff
@@ -1,4 +1,9 @@
+import { toJson } from "xml2json";
 export const extractClientKey = (text: string): string | undefined  => {
-	const [, plainClientKey] = /<key>([0-9a-z-:]+)<\/key>/gmi.exec(text) || [undefined, undefined];
-	return plainClientKey;
+	try {
+		const json = toJson(text, { object: true });
+		return (json.consumer as { [key: string]: string }).key;
```

Comment:
Not sure why you need to set the type?  Doesn't the `toJson` just return `any`?

Reply:
Oh, you're using `@types/xml2json`.  That definition is horrible and not genericized.  I'd just remove the dependency and just do `return json?.consumer?.key;`

Reply:
Make sense. Fixed.